client/my-sites/email/index.js

```diff
@@ -166,4 +166,9 @@ export default function () {
 		],
 		handlers: [ ...commonHandlers, controller.emailManagementForwarding, makeLayout, clientRender ],
 	} );
+
+	registerMultiPage( {
+		paths: [ paths.inboxManagement() ],
```

Comment:
You need to register all the path/url possibilities. This will bring up the site-selector because no site is specified. 
You need to also register a site-enabled version of the url i.e.
`/email/inbox/<site-name>`